### PR TITLE
Rest api for blog, the blogs endpoint crud done

### PR DESCRIPTION
**Pull request by NG-Solennel**
added the comment and like endpoints

setting things up almost done \  finessing likes addition and removal

### CODE REVIEW

**Comment by NG-Solennel** [comment]
So the second commit is for a github commit for merging

File: .env [hunk 1]
```diff
@@ -0,0 +1,14 @@
+PORT = 3000
+NODE_ENV = "development"
```
Comment (Eli250) [inline .env]: Why did you push your `.env` file? Don't push it.
It has to be added to `.gitingore`

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import mongoose from "mongoose";
+import morgan from "morgan";
+import "dotenv/config";
+import routes from "./routes";
+import session from "express-session";
+import passport from "passport";
+import passportGoogle from "./middleware/auth/passportGoogle";
+import passportSetup from "./middleware/auth/passport";
+const app = express();
+app.use(session({ secret: "andela", resave: false, saveUninitialized: true }));
```
Comment (Eli250) [inline src/app.js:11]: Put your secret key in env variables

File: src/controller/blogController.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      const data = {
+        title,
+        content,
+        image: {
+          public_id: imgResult.public_id,
+          url: imgResult.secure_url,
+        },
+      };
+      const response = await BlogServices.createBlog(data);
+      if (response.type == "error") {
```
Comment (Eli250) [inline src/controller/blogController.js:22]: Move this into validation middleware.

File: src/controller/likeController.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        return res.status(400).json({ ValidationError: response.data });
+      } else if (response.type == "response") {
+        res.status(200).json({ likedBlog: response.data });
+      }
+    } catch (error) {
+      console.log(error);
+      res.status(500).json({ error: error });
+    }
+  }
+  // static async removeLike(req, res) {
```
Comment (Eli250) [inline src/controller/likeController.js]: Remove these comments.

File: src/controller/likeController.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { LikeServices } from "../services/likeService";
+export class LikeController {
+  static async like(req, res) {
+    try {
+      const response = await LikeServices.like(req.params.id, req.body);
+
+      if (response.type == "error") {
```
Comment (Eli250) [inline src/controller/likeController.js:6]: Move this to validation middleware.

File: src/controller/messageController.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      const { name, email, description, hiring, message } = req.body;
+      const data = {
+        name,
+        email,
+        description,
+        hiring,
+        message,
+      };
+      const response = await MessageServices.sendMessage(data);
+      if (response.type == "error") {
```
Comment (Eli250) [inline src/controller/messageController.js:16]: Use a middleware

File: src/controller/userController.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  static async login(req, res) {
+    try {
+      const { email, password } = req.body;
+      const data = {
+        email,
+        password,
+      };
+      const response = await UserServices.login(data);
```
Comment (Eli250) [inline src/controller/userController.js:34]: Use validation middleware.

**Review by Eli250 — CHANGES_REQUESTED** [review]
Work on the comments.

**Pull request closed without merging**